test/controllers/products_controller_test.rb

```diff
@@ -26,13 +26,66 @@
     end
   end
 
-  it "must get the show view" do
-    get product_path(one.id)
-    must_respond_with :success
+  describe "show" do
+    it "must get the show view" do
+      get product_path(one.id)
+      must_respond_with :success
+
+      get product_path(-1)
+      must_respond_with :redirect
+      must_redirect_to root_path
+    end
+  end
+
+  describe "new product" do
+    it "should get new if a mercant is signed in" do
+      @merchant = merchants(:sappy1)
+      login(@merchant, :github)
+      get new_product_path
+      must_respond_with :success
+    end
+
+    it "must redirect to root path if a merchant is not signed in" do
+      # With no mercant signed in
+      get edit_product_path(one.id)
+      must_respond_with :redirect
+      flash[:status].must_equal :error
+      must_redirect_to root_path
+    end
+  end
+
+  describe "create product" do
+    before do
+      @merchant = merchants(:sappy1)
+      login(@merchant, :github)
+    end
+
+    it "should be able to successfully create a new product" do
+        proc {
+          post products_path, params: { name: "newbie", price: 2, category: "new category", quantity: 1, merchant_id: @merchant.id }
+        }.must_change 'Product.count', 1
+      must_respond_with :redirect
+    end
+
+    it "should not create the product if it is missing a name" do
+      proc {
+        post products_path, params: { name: "", price: 2, category: "new category", quantity: 1, merchant_id: merchants(:sappy1).id }
+      }.must_change 'Product.count', 0
+
+      must_respond_with :redirect
+      flash[:status].must_equal :error
+    end
+
+    it "should not create the product if it is missing a price" do
+      proc {
+        post products_path, params: { name: "Name", category: "new category", quantity: 1, merchant_id: merchants(:sappy1).id }
+      }.must_change 'Product.count', 0
+
+      must_respond_with :redirect
+      flash[:status].must_equal :error
+    end
 
-    get product_path(-1)
-    must_respond_with :redirect
-    must_redirect_to root_path
+    #Need to test other non-validated values or is it covered enough in the model??
```

Comment:
Technically I think we should test those things, but as long as they are tested in the model we should be okay.